sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala

```diff
@@ -178,8 +179,17 @@ private[parquet] class ParquetRowConverter(
 
   // Converters for each field.
   private[this] val fieldConverters: Array[Converter with HasParentContainerUpdater] = {
+    val catalystFieldNameToIndex = if (caseSensitive) {
+      catalystType.fieldNames.zipWithIndex.toMap
+    } else {
+      CaseInsensitiveMap(catalystType.fieldNames.zipWithIndex.toMap)
+    }
     parquetType.getFields.asScala.map { parquetField =>
-      val fieldIndex = catalystType.fieldIndex(parquetField.getName)
+      val fieldIndex = catalystFieldNameToIndex.getOrElse(parquetField.getName,
+        throw new IllegalArgumentException(
```

Comment:
It seems that we cannot avoid a runtime exception here. BTW, although I understand that this follows the existing `IllegalArgumentException` from `org.apache.spark.sql.types.StructType`, `RuntimeException` might be better in this context like the other exceptions in this class. (cc @cloud-fan )